### PR TITLE
Vibhansa/listapiblock

### PR DESCRIPTION
**Pull request by vibhansa-msft**
Block listing operation on mount for configured amount of seconds.

### CODE REVIEW

**Review by NaraVen — COMMENTED** [review]
conflicts need to be resolved here

File: blobfuse/src/BlobfuseGcCache.cpp [hunk 1]
```diff
@@ -184,7 +188,7 @@ void gc_cache::run_gc_cache()
         else
         {
             // no file was timed out - let's wait a second
-            usleep(1000);
```
Comment (NaraVen) [inline blobfuse/src/BlobfuseGcCache.cpp:187]: is this related to not listing on mount?

File: blobfuse/include/BlobfuseGlobals.h [hunk 2]
```diff
@@ -84,6 +85,7 @@ struct cmdlineOptions
     const char *help; // print blobfuse usage
     const char *concurrency; // Max Concurrency factor for blob client wrapper (default 40)
     const char *cache_size_mb; // MAX Size of cache in MBs
+    const char *block_list_on_mount_for_secs; // Block the list api call on mount for n seconds
```
Comment (NaraVen) [inline blobfuse/include/BlobfuseGlobals.h:88]: the purpose is to not do the default list so why use time here?

**Review by NaraVen — CHANGES_REQUESTED** [review]

File: blobfuse/include/BlobfuseGlobals.h (hunk 2 above)
Comment (vibhansa-msft) [reply blobfuse/include/BlobfuseGlobals.h:88]: we need to block the list call for X seconds post mount. We do not have any way to determine its an implicit list by fuse driver or user driven operation, so blocking all list for few seconds after mount.

File: blobfuse/src/BlobfuseGcCache.cpp (hunk 1 above)
Comment (vibhansa-msft) [reply blobfuse/src/BlobfuseGcCache.cpp:187]: this was one ms sleep was was converted to 1 sec sleep that time. Now we need to make this configurable.

File: blobfuse/include/BlobfuseGlobals.h (hunk 2 above)
Comment (NaraVen) [reply blobfuse/include/BlobfuseGlobals.h:88]: blocking alone will not be enough, after mount it should not list at all.

Comment (vibhansa-msft) [reply blobfuse/include/BlobfuseGlobals.h]: Updated.

**Review by NaraVen — APPROVED** [review]
